packages/cli/src/utils/input.ts

```diff
@@ -46,18 +48,24 @@ function quoteArguments(args: string) {
   return args;
 }
 
-export function parseArg(input: string | string[], type: string): any {
+export function parseArg(input: string | string[], { type, components }: Pick<MethodArg, 'type' | 'components'>): any {
```

Comment:
Why not use the entire `MethodArg` type here? I imagine that using `Pick` simplifies the tests? Is there another reason?

If we insist on not using the entire type, I think we should define a type alias in this file. It represents an important concept, and it's used twice in the file.

Reply:
The problem was in the recursive call, since the types for an array (for example) are not named. I agree with giving it a proper name, just created `MethodArgType` for that.